Sources/ClientRuntime/Interceptor/HttpInterceptorProvider.swift

```diff
@@ -13,5 +16,5 @@ public protocol HttpInterceptorProvider {
     /// Creates an instance of an `HttpInterceptor` implementation.
     ///
     /// - Returns: The `HttpInterceptor` implementation.
-    func create<InputType, OutputType>() -> any HttpInterceptor<InputType, OutputType>
+    func create<InputType, OutputType>() -> any Interceptor<InputType, OutputType, HTTPRequest, HTTPResponse>
```

Comment:
Have this method return a plain Interceptor, with the return & request types pinned to HTTP.
This exactly matches the type that an Interceptors instance stores.